waveorder/io/reader.py

```diff
@@ -202,9 +329,254 @@ def shape(self):
+        Strategy:
+            1. Gather summary metadata from any metadata.txt
+            2. Build a map between image coordinates and image file names
+            3. Upon call to extract data, assign each scene to self.positions (similar to OmeTiffReader)
+
+        :param folder: str
+            folder containing position subdirectories, which contain singlepage tiff sequences
+        :param extract_data: bool
+            True if zarr arrays should be extracted immediately
+        """
+
+        if not os.path.isdir(folder):
+            raise NotImplementedError("supplied path for singlepage tiff sequence reader is not a folder")
+
+        self.log = logging.getLogger(__name__)
+        self.positions = {}
+        self.mm_meta = None
+        self.stage_positions = 0
+        self.height = 0
+        self.width = 0
+        self.frames = 0
+        self.slices = 0
+        self.channels = 0
+        self.channel_names = []
+
+        self.coord_to_filename = {}
+
+        # identify type of subdirectory
+        sub_dirs = self._get_sub_dirs(folder)
+        if sub_dirs:
+            pos_path = os.path.join(folder, sub_dirs[0])
+        else:
+            raise AttributeError("supplied folder does not contain position or default subdirectories")
+
+        # pull one metadata sample and extract experiment dimensions
+        metadata_path = os.path.join(pos_path, 'metadata.txt')
+        with open(metadata_path, 'r') as f:
+            self.mm_meta = json.load(f)
+
+        self.mm_version = self.mm_meta['Summary']['MicroManagerVersion']
+        if self.mm_version == '1.4.22':
+            self._mm1_meta_parser()
+        elif 'beta' in self.mm_version:
+            self._mm2beta_meta_parser()
+        elif 'gamma' in self.mm_version:
+            self._mm2gamma_meta_parser()
+        else:
+            raise NotImplementedError(
+                f'Current MicroManager reader only supports version 1.4.22 and 2.0 but {self.mm_version} was detected')
+
+        # create coordinate to filename maps
+        self.coord_to_filename = self.read_tiff_series(folder)
+
+        if extract_data:
+            self._create_stores()
+
+    def get_zarr(self, position_):
+        if not self.positions:
+            self._create_stores()
+        return self.positions[position_]
+
+    def get_array(self, position_):
+        if not self.positions:
+            self._create_stores()
+        return np.array(self.positions[position_])
+
+    def get_num_positions(self):
+        if self.positions:
+            return len(self.positions)
+        else:
+            self.log.error("singlepage tiffs not loaded")
+
+    def _create_stores(self):
+        """
+        extract all singlepage tiffs at each coordinate and place them in a zarr array
+        coordinates are of shape = (pos, time, channel, z)
+        :return:
+        """
+        self.log.info("")
+        z = zarr.zeros(shape=(self.frames,
```

Comment:
In the OME Tiff reader, the zarr stores created are of the dimension (C, Z, Y, X).  Here we have created stores of (P, C, Z, Y, X) dimensions, with the P dimensions always equal to 1.  I think that we should remove this dimension to keep it consistent with the other reader and because it is implied when you call a certain position with `self.get_array(pos)`

Reply:
good catch.  But to clarify -- the coordinate is (P, T, C, Z, Y, X).  Here, zarr is opening a (T, C, Z, Y, X) and placing it in `self.positions` at P

Reply:
I see, this may be okay then.  Maybe we should then expand the ometiff reader to always include a time dimension.  Or here we can only include the time dimension when it is present, as I believe is the case for the ometiff reader

Reply:
For reconstruction, it is very useful if the raw data is in consistent dimensions. For example, if you'd like to do a z projection you can simply average along the third dimensions without having to worry how many channels or time points were in this specific dataset. I'd suggest modifying the reader such that data is always returned in (T, C, Z, Y, X) dimensions order, padding with size=1 dimensions where necessary.

I believe this is also what aicsimageio does. The OME tif metadata contains information on the size of each dimension.

Reply:
@ieivanov Ok this should be fixed in now.  The dimensions are now consistently (T, C, Z, Y, X) when reading ome-tiff